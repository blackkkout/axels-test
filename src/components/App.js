import { createTheme, ThemeProvider, CssBaseline } from '@mui/material';

import { Root } from '../styled/App';
import { palette } from '../styled/config';
import { Order } from './Order';

const theme = createTheme({
  palette,
});

export const App = () => (
  <ThemeProvider theme={theme}>
    <CssBaseline />
    <Root>
      <Order />
    </Root>
  </ThemeProvider>
);
