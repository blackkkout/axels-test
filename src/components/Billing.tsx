import { useState } from 'react';
import {
  Button,
  FormControl,
  FormLabel,
  Stack,
  Typography,
  TextField,
  Autocomplete,
  InputAdornment,
  IconButton,
  FormHelperText,
  OutlinedInput,
  Link,
  CircularProgress,
} from '@mui/material';
import { LocationSearching } from '@mui/icons-material';
import { useFormik } from 'formik';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import * as yup from 'yup';

import { getAddress } from '../api/address';
import { geolocationSelector } from '../redux/ducks/geolocation';

const validationSchema = yup.object().shape({
  fullName: yup.string().required('Full Name is required'),
  email: yup
    .string()
    .email('Email must be a valid email')
    .required('Email Address is required'),
  streetAddress: yup.string().required('Street Address is required'),
  apt: yup.string(),
  city: yup.string().required('City is required'),
  country: yup.string().required('Country is required'),
  zip: yup.string().required('ZIP is required'),
});

export const Billing = () => {
  const navigate = useNavigate();
  const [isAddressLoading, setIsAddressLoading] = useState(false);

  const formik = useFormik({
    initialValues: {
      fullName: '',
      email: '',
      streetAddress: '',
      apt: '',
      city: '',
      country: '',
      zip: '',
    },
    validationSchema,
    onSubmit: (_) => {
      navigate('/payment');
    },
  });

  const countries = ['United States', 'Canada', 'Mexico', 'Ukraine'];

  const coords = useSelector(geolocationSelector);

  const handleGetAddress = async () => {
    if (coords && coords.latitude && coords.longitude) {
      try {
        setIsAddressLoading(true);
        const data = await getAddress(coords.latitude, coords.longitude);

        formik.setFieldValue('streetAddress', data.address.road || '');
        formik.setFieldValue('apt', data.address.house_number || '');
        formik.setFieldValue(
          'city',
          data.address.city || data.address.town || '',
        );
        formik.setFieldValue('country', data.address.country || '');
        formik.setFieldValue('zip', data.address.postcode || '');
      } catch (error) {
        console.error(error);
      } finally {
        setIsAddressLoading(false);
      }
    }
  };

  return (
    <form onSubmit={formik.handleSubmit}>
      <Stack
        direction="row"
        justifyContent="space-between"
        alignItems="flex-end"
        marginTop={3}
        marginBottom={2}
      >
        <Typography variant="h5" color="primary.main">
          Billing Information
        </Typography>
        <Link whiteSpace="nowrap" fontSize={12}>
          Same as shipping
        </Link>
      </Stack>
      <Stack spacing={2} marginBottom={2}>
        <Stack spacing={1}>
          <FormControl fullWidth>
            <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
              Billing Contact
            </FormLabel>
            <TextField
              name="fullName"
              placeholder="Full Name"
              size="small"
              fullWidth
              value={formik.values.fullName}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={formik.touched.fullName && Boolean(formik.errors.fullName)}
              helperText={formik.touched.fullName && formik.errors.fullName}
            />
          </FormControl>
          <FormControl fullWidth>
            <TextField
              name="email"
              placeholder="Email Address"
              size="small"
              fullWidth
              value={formik.values.email}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={formik.touched.email && Boolean(formik.errors.email)}
              helperText={formik.touched.email && formik.errors.email}
            />
          </FormControl>
        </Stack>
        <Stack spacing={1}>
          <FormControl fullWidth>
            <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
              Billing Address
            </FormLabel>
            <TextField
              name="streetAddress"
              placeholder="Street Address"
              size="small"
              fullWidth
              value={formik.values.streetAddress}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={
                formik.touched.streetAddress &&
                Boolean(formik.errors.streetAddress)
              }
              helperText={
                formik.touched.streetAddress && formik.errors.streetAddress
              }
            />
          </FormControl>
          <FormControl fullWidth>
            <TextField
              name="apt"
              placeholder="Apt, Suite, Bldg, Gate Code. (optional)"
              size="small"
              fullWidth
              value={formik.values.apt}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={formik.touched.apt && Boolean(formik.errors.apt)}
              helperText={formik.touched.apt && formik.errors.apt}
            />
          </FormControl>
        </Stack>
        <FormControl fullWidth>
          <OutlinedInput
            name="city"
            size="small"
            endAdornment={
              <InputAdornment position="end">
                <IconButton
                  aria-label="get location"
                  edge="end"
                  onClick={handleGetAddress}
                  disabled={isAddressLoading || !coords}
                >
                  {isAddressLoading ? (
                    <CircularProgress size={24} />
                  ) : (
                    <LocationSearching />
                  )}
                </IconButton>
              </InputAdornment>
            }
            placeholder="City"
            value={formik.values.city}
            onChange={formik.handleChange}
            onBlur={formik.handleBlur}
            error={formik.touched.city && Boolean(formik.errors.city)}
          />
          {formik.touched.city && formik.errors.city && (
            <FormHelperText sx={{ color: 'error.main' }}>
              {formik.errors.city}
            </FormHelperText>
          )}
        </FormControl>
        <Stack spacing={2} direction="row">
          <FormControl fullWidth>
            <Autocomplete
              freeSolo
              disablePortal
              options={countries}
              value={formik.values.country}
              getOptionLabel={(option) => option}
              onChange={(_, value) => formik.setFieldValue('country', value)}
              renderInput={(params) => (
                <TextField
                  {...params}
                  name="country"
                  size="small"
                  placeholder="Country"
                  onBlur={formik.handleBlur}
                  error={
                    formik.touched.country && Boolean(formik.errors.country)
                  }
                  helperText={formik.touched.country && formik.errors.country}
                />
              )}
            />
          </FormControl>
          <FormControl fullWidth>
            <TextField
              name="zip"
              placeholder="ZIP"
              size="small"
              fullWidth
              value={formik.values.zip}
              onChange={formik.handleChange}
              onBlur={formik.handleBlur}
              error={formik.touched.zip && Boolean(formik.errors.zip)}
              helperText={formik.touched.zip && formik.errors.zip}
            />
          </FormControl>
        </Stack>
      </Stack>
      <Button type="submit" variant="contained" fullWidth={false}>
        Continue
      </Button>
    </form>
  );
};
