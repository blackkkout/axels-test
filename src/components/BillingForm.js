import {
  Button,
  FormControl,
  FormLabel,
  Stack,
  Typography,
  TextField,
  Autocomplete,
  InputAdornment,
  IconButton,
  OutlinedInput,
  Link,
} from '@mui/material';
import { LocationSearching } from '@mui/icons-material';

export const BillingForm = ({ onSubmit }) => {
  const countries = ['United States', 'Canada', 'Mexico', 'Ukraine'];

  const handleSubmit = (event) => {
    event.preventDefault();
    onSubmit();
  };

  return (
    <form onSubmit={handleSubmit}>
      <Stack
        direction="row"
        justifyContent="space-between"
        alignItems="flex-end"
        marginTop={3}
        marginBottom={2}
      >
        <Typography variant="h5" color="primary.main">
          Billing Information
        </Typography>
        <Link whiteSpace="nowrap" fontSize={12}>
          Same as shipping
        </Link>
      </Stack>
      <Stack spacing={2} marginBottom={2}>
        <Stack spacing={1}>
          <FormControl fullWidth>
            <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
              Billing Contact
            </FormLabel>
            <TextField placeholder="Full Name" size="small" fullWidth />
          </FormControl>
          <FormControl fullWidth>
            <TextField placeholder="Email Address" size="small" fullWidth />
          </FormControl>
        </Stack>
        <Stack spacing={1}>
          <FormControl fullWidth>
            <FormLabel sx={{ marginBottom: 1, color: 'primary.main' }}>
              Billing Address
            </FormLabel>
            <TextField placeholder="Street Address" size="small" fullWidth />
          </FormControl>
          <FormControl fullWidth>
            <TextField
              placeholder="Apt, Suite, Bldg, Gate Code. (optional)"
              size="small"
              fullWidth
            />
          </FormControl>
        </Stack>
        <FormControl fullWidth>
          <OutlinedInput
            size="small"
            endAdornment={
              <InputAdornment position="end">
                <IconButton aria-label="get location" edge="end">
                  <LocationSearching />
                </IconButton>
              </InputAdornment>
            }
            placeholder="City"
          />
        </FormControl>
        <Stack spacing={2} direction="row">
          <FormControl fullWidth>
            <Autocomplete
              disablePortal
              options={countries}
              renderInput={(params) => (
                <TextField {...params} size="small" placeholder="Country" />
              )}
            />
          </FormControl>
          <FormControl fullWidth>
            <TextField placeholder="ZIP" size="small" fullWidth />
          </FormControl>
        </Stack>
      </Stack>
      <Button type="submit" variant="contained" fullWidth={false}>
        Continue
      </Button>
    </form>
  );
};
