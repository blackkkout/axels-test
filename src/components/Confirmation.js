import { Link, Stack, Typography } from '@mui/material';

export const Confirmation = () => (
  <Stack direction="column" gap={3}>
    <Typography variant="h5" color="primary.main">
      Thank you for your order!
    </Typography>
    <Stack direction="column" gap={1}>
      <Typography fontWeight={600}>Order number is: 123456789</Typography>
      <Typography lineHeight={1.25}>
        You will receive an email confirmation shortly to{' '}
        <Link>jonathan.smith@gmail.com</Link>
      </Typography>
    </Stack>
    <Typography lineHeight={1.25}>
      Estimated delivery day is <br />
      <b>Friday, July 15, 2022</b>
    </Typography>
    <Link fontSize={13}>Print Recipe</Link>
  </Stack>
);
