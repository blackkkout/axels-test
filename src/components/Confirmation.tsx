import { Link, Stack, Typography } from '@mui/material';
import { useRef } from 'react';
import { useReactToPrint } from 'react-to-print';

export const Confirmation = () => {
  const componentRef = useRef<HTMLDivElement>(null);
  const handlePrint = useReactToPrint({
    content: () => componentRef.current,
  });

  return (
    <div>
      <div ref={componentRef}>
        <Stack direction="column" gap={3}>
          <Typography
            variant="h5"
            color="primary.main"
            sx={{
              '@media print': {
                color: 'black',
              },
            }}
          >
            Thank you for your order!
          </Typography>
          <Stack direction="column" gap={1}>
            <Typography fontWeight={600}>Order number is: 123456789</Typography>
            <Typography lineHeight={1.25}>
              You will receive an email confirmation shortly to{' '}
              <Link
                sx={{
                  '@media print': {
                    color: 'black',
                  },
                }}
              >
                jonathan.smith@gmail.com
              </Link>
            </Typography>
          </Stack>
          <Typography lineHeight={1.25}>
            Estimated delivery day is <br />
            <b>Friday, July 15, 2022</b>
          </Typography>
        </Stack>
      </div>
      <Link fontSize={13} sx={{ cursor: 'pointer' }} onClick={handlePrint}>
        Print Recipe
      </Link>
    </div>
  );
};
