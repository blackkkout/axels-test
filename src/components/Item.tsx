import { Stack, Typography, Box } from '@mui/material';

import { Image } from '../styled/Item';
import { Order } from '../types/order';

interface ItemProps {
  order: Order;
}

export const Item = ({ order }: ItemProps) => (
  <Stack marginBottom={1} direction="row" justifyContent="space-between">
    <Stack direction="row">
      <Box width={50} height={50} marginRight={1}>
        <Image src={order.image} alt={order.name} />
      </Box>
      <Stack direction="column" fontSize={11}>
        <Typography fontSize="inherit" color="grey.600">
          {order.name}
        </Typography>
        <Typography fontSize="inherit">
          {order.shape}
          <br />
          Quantity: {order.quantity}
        </Typography>
      </Stack>
    </Stack>
    <Typography fontSize={11} color="grey.600">
      {order.price}
    </Typography>
  </Stack>
);
