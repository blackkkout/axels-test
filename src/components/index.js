export { App } from './App';
export { BillingForm } from './BillingForm';
export { Order } from './Order';
export { OrderConfirmation } from './OrderConfirmation';
export { OrderItem } from './OrderItem';
export { OrderSummary } from './OrderSummary';
export { PaymentForm } from './PaymentForm';
export { ShippingForm } from './ShippingForm';
