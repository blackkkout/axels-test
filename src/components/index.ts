export { Billing } from './Billing';
export { Order } from './Order';
export { Confirmation } from './Confirmation';
export { Item } from './Item';
export { Summary } from './Summary';
export { Payment } from './Payment';
export { Shipping } from './Shipping';
