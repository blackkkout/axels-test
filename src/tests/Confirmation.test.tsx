import { render } from '@testing-library/react';

import { Confirmation } from '../components/Confirmation';

describe('Confirmation', () => {
  it('should take a snapshot', () => {
    const view = render(<Confirmation />);
    expect(view).toMatchSnapshot();
  });
});
