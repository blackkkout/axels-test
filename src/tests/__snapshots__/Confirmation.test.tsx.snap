// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation should take a snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div
            class="MuiStack-root css-msk5bl-MuiStack-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-1k1sedy-MuiTypography-root"
            >
              Thank you for your order!
            </h5>
            <div
              class="MuiStack-root css-hwnj0i-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
              >
                Order number is: 123456789
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-lh1nwc-MuiTypography-root"
              >
                You will receive an email confirmation shortly to
                 
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-mb785f-MuiTypography-root-MuiLink-root"
                >
                  jonathan.smith@gmail.com
                </a>
              </p>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-lh1nwc-MuiTypography-root"
            >
              Estimated delivery day is 
              <br />
              <b>
                Friday, July 15, 2022
              </b>
            </p>
          </div>
        </div>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-31txpq-MuiTypography-root-MuiLink-root"
        >
          Print Recipe
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div
          class="MuiStack-root css-msk5bl-MuiStack-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1k1sedy-MuiTypography-root"
          >
            Thank you for your order!
          </h5>
          <div
            class="MuiStack-root css-hwnj0i-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1qqynwt-MuiTypography-root"
            >
              Order number is: 123456789
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-lh1nwc-MuiTypography-root"
            >
              You will receive an email confirmation shortly to
               
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-mb785f-MuiTypography-root-MuiLink-root"
              >
                jonathan.smith@gmail.com
              </a>
            </p>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-lh1nwc-MuiTypography-root"
          >
            Estimated delivery day is 
            <br />
            <b>
              Friday, July 15, 2022
            </b>
          </p>
        </div>
      </div>
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-31txpq-MuiTypography-root-MuiLink-root"
      >
        Print Recipe
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
